config.py

```diff
@@ -0,0 +1,70 @@
+parser.add_argument('-v', '--version', type=str, default='v0.14.1', help='Spack version, Default: ' + spack_version)
+parser.add_argument('-r', '--reposdir', type=str, help='repos.yaml install directory')
+parser.add_argument('-p', '--pckgidir', type=str, help='Define spack package, modules & stages installation directory. Default: tsa; /scratch/$USER/spack, daint; /scratch/snx3000/$USER/spack')
+args=parser.parse_args()
+
+if args.machine is None:
+    print('Error: machine name required!')
+    exit()
+  
+if args.idir is not None:
+    if not os.path.isdir(args.idir + '/spack'):
+        print('Cloning spack instance to:' + args.idir)
+        if args.version is None:
+            args.version = spack_version
+        clone_cmd = 'git clone git@github.com:spack/spack.git -b' + args.version + ' ' + args.idir + '/spack'   
+        os.system(clone_cmd)
+
+print('Installing mch packages & ' + args.machine + ' config files')
+
+if args.reposdir is not None:
+    if os.path.isdir(args.reposdir) and not os.path.isfile(args.reposdir + '/repos.yaml'):
+        repos_data = yaml.load(open('./sysconfigs/repos.yaml', 'r+'))
+        repos_data['repos'] = args.reposdir
+        yaml.dump(repos_data, open('./sysconfigs/repos.yaml', 'r+'))
+        print('Installing repos.yaml on ' + args.reposdir)
+        os.popen('cp ' + dir_path + '/sysconfigs/repos.yaml ' + args.reposdir)
+
+# configure config.yaml
+
+config_data = yaml.load(open('sysconfigs/config.yaml', 'r+'))
+
+if args.pckgidir is None:
+    if 'admin' in args.machine:
+        args.pckgidir = '/project/g110'
+    elif args.machine == 'tsa':
+        args.pckgidir = '/scratch/$user/spack'
+    elif args.machine == 'daint':
+        args.pckgidir = '/scratch/snx3000/$user/spack'
+      
+config_data['config']['install_tree'] = args.pckgidir + '/spack-install/' + args.machine.replace('admin-', '')
+config_data['config']['build_stage'] = args.pckgidir + '/spack-stages/' + args.machine.replace('admin-', '')
+config_data['config']['module_roots']['tcl'] = args.pckgidir + '/modules'
+yaml.dump(config_data, open('./sysconfigs/config.yaml', 'r+'))
+
+# copy modified config.yaml file in site scope of spack instance
+os.popen('cp -rf sysconfigs/config.yaml ' + args.idir + '/spack/etc/spack')
+
+# copy modified upstreams.yaml if not admin
+if not 'admin' in args.machine:
+    upstreams_data = yaml.load(open('./sysconfigs/upstreams.yaml', 'r+'))
+    upstreams_data['upstreams']['spack-instance-1']['install_tree'] = '/project/g110/spack-install/' + args.machine.replace('admin-', '')
+    yaml.dump(upstreams_data, open('./sysconfigs/upstreams.yaml', 'r+'))
+    os.popen('cp -rf sysconfigs/upstreams.yaml ' + args.idir + '/spack/etc/spack')
+
+# copy modules.yaml, packages.yaml and compiles.yaml files in site scope of spack instance
+os.popen('cp -rf sysconfigs/' + args.machine.replace('admin-', '') + '/* ' +  args.idir + '/spack/etc/spack')
```

Comment:
any reason for using sometimes os.system, and others os.popen?

Reply:
Yeah true gonna change everything to os.popen, forgot I mixed them, thanks!

Reply:
Actually, I had to revert it because git is not able to lock its config file with os.popen(), so if it is fine I'll stay with the actual solution.